local/build-binary

```diff
@@ -61,6 +62,38 @@ fi
 
 wget_loop "googletest-release-${GMOCK_VERSION}.zip" "https://github.com/google/googletest/archive/release-${GMOCK_VERSION}.zip"
 
+if [[ ${WITH_BORINGSSL} == "true" ]]; then
+    # ------------------------------------------------------------------------------
+    # compile boringssl libs
+    # ------------------------------------------------------------------------------
+    pushd ${WORKDIR}
+    git clone https://github.com/google/boringssl.git boringssl_sources
+    mkdir boringssl_sources/bin
+    cd boringssl_sources/bin
+    if [[ -f /etc/redhat-release ]]; then
+        RHVER="$(rpm --eval %rhel)"
+        case "${RHVER}" in
+            7|8)
+                if [[ ${RHVER} == 7 ]]; then
+                    source /opt/rh/go-toolset-7/enable
+                fi
+                source /opt/rh/devtoolset-8/enable
+                BORING_SSL_CMAKE=/usr/bin/cmake3
+            ;;
+            6) echo "Unsupported CentOS version" && exit 1 ;;
+        esac
+    else
+        BORING_SSL_CMAKE=/usr/bin/cmake
+    fi
+    ${BORING_SSL_CMAKE} .. -DCMAKE_CXX_FLAGS="-fPIC" -DCMAKE_C_FLAGS="-fPIC"
+    if make -j$(nproc); then
```

Comment:
Should we fail if there was a problem building BoringSSL?

Reply:
Yes it should fail in case of compilation failure, I'll add `else; exit 1`

Reply:
re: Added in 7d02b5e